src/main/java/Ui.java

```diff
@@ -0,0 +1,183 @@
+import java.util.Scanner;
+
+/**
+ * Ui class is the main processor for users inputs,
+ * Ui takes in user inputs, categorises it and stores it.
+ */
+public class Ui {
```

Comment:
Is it possible to further abstract the code? This style of deep nesting is not encouraged and you might have to organize the code again.